test/test_utils.py

```diff
@@ -1198,6 +1219,68 @@ def test_build_eager(self):
         for message, value, expected in test_data:
             self.assertEqual(expected, build_eager(value), message)
 
+    def test_compute_hash(self):
+        self.assertEqual("4de279c82056603e91aab3930a593b8b887d9e48",
+                         compute_hex_hash("https://cloudinary.com/images/old_logo.png"))
+
+        original_value = str(uuid.uuid4())
+
+        self.assertEqual(compute_hex_hash(original_value), compute_hex_hash(original_value),
+                         "Equal inputs should be hashed the same way")
+
+        self.assertNotEqual(compute_hex_hash(original_value), compute_hex_hash("some string"),
+                            "Unequal inputs hashes should not match")
+
+    def test_verify_api_response_signature(self):
+        public_id = 'tests/logo.png'
+        test_version = 1
+        api_response_signature = '08d3107a5b2ad82e7d82c0b972218fbf20b5b1e0'
+
+        cloudinary.config(api_secret='X7qLTrsES31MzxxkxPPA-pAGGfU')
+
+        self.assertTrue(verify_api_response_signature(public_id, test_version, api_response_signature),
+                        "The response signature is valid for the same parameters")
+
+        self.assertFalse(verify_api_response_signature(public_id, test_version + 1, api_response_signature),
+                         "The response signature is invalid for the wrong version")
+
+        cloudinary.config(api_secret='b')
```

Comment:
That is a bad practice, in case assertion fails, we never get to this line, use `with patch()`  for setting `api_key`